modules/widgets/follow-button.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+add_action( 'widgets_init', 'follow_button_register_widget' );
+function follow_button_register_widget() {
+	register_widget( 'Follow_Button_Widget' );
+}
+
+class Follow_Button_Widget extends WP_Widget {
+
+	public function __construct() {
+		parent::__construct(
+			'follow_button_widget',
+			__( 'Follow Button', 'jetpack' ),
+			array( 'description' => __('Add a WordPress.com follow button to allow people to follow your blog easier', 'jetpack' ) )
+		);
+	}
+
+	public function widget( $args, $instance ) {
+		$attributes = array();
+		$instance = wp_parse_args( (array) $instance, array( 'show_name' => 1, 'show_count' => 0 ) );
+
+		if ( empty( $instance['show_name'] ) )
+			$attributes[] = 'data-show-blog-name="false"';
+
+		if ( ! empty( $instance['show_count'] ) )
+			$attributes[] = 'data-show-follower-count="true"';
+
+		echo $args['before_widget'];
+		?>
+
+		<a class="wordpress-follow-button" href="<?php echo site_url(); ?>" data-blog="<?php echo site_url(); ?>" data-lang="<?php echo get_locale(); ?>" <?php if ( ! empty( $attributes ) ) echo implode( ' ', $attributes ); ?>>Follow <?php bloginfo( 'name' ); ?> on WordPress.com</a>
```

Comment:
Needs URL escaping, attribute escaping and making string available for translation.